dace/transformation/dataflow/redundant_array.py

```diff
@@ -558,8 +558,12 @@ def apply(self, graph, sdfg):
             wcr = e1.data.wcr
             wcr_nonatomic = e1.data.wcr_nonatomic
             for e3 in path:
+                if e3.data.data != in_array.data:
```

Comment:
Something to check is whether `other_subset` exists. That still has to be updated.